streams/src/main/java/org/apache/kafka/streams/processor/internals/tasks/DefaultTaskManager.java

```diff
@@ -118,6 +120,41 @@ public StreamTask assignNextTask(final TaskExecutor executor) {
         });
     }
 
+    @Override
+    public void awaitProcessableTasks() throws InterruptedException {
+        final boolean interrupted = returnWithTasksLocked(() -> {
+            for (final Task task : tasks.activeTasks()) {
+                if (!assignedTasks.containsKey(task.id()) &&
+                    !lockedTasks.contains(task.id()) &&
+                    canProgress((StreamTask) task, time.milliseconds())
+                ) {
+                    log.info("Await unblocked: returning early from await since a processable task {} was found", task.id());
```

Comment:
I think info might be too high of a log level. This should be called frequently right?

Reply:
Yes, you are right. I changed it in a later PR, but I pulled it back into this PR.